service-base/src/main/java/org/eclipse/hono/service/credentials/EventBusCompleteCredentialsAdapter.java

```diff
@@ -0,0 +1,317 @@
+/*******************************************************************************
+ * Copyright (c) 2016, 2019 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package org.eclipse.hono.service.credentials;
+
+import java.net.HttpURLConnection;
+import java.util.Objects;
+import java.util.Optional;
+
+import org.eclipse.hono.auth.BCryptHelper;
+import org.eclipse.hono.auth.HonoPasswordEncoder;
+import org.eclipse.hono.client.ClientErrorException;
+import org.eclipse.hono.tracing.TracingHelper;
+import org.eclipse.hono.util.CredentialsConstants;
+import org.eclipse.hono.util.CredentialsObject;
+import org.eclipse.hono.util.CredentialsResult;
+import org.eclipse.hono.util.EventBusMessage;
+import org.eclipse.hono.util.MessageHelper;
+
+import io.opentracing.Span;
+import io.vertx.core.Future;
+import io.vertx.core.json.JsonObject;
+
+/**
+ * A base class for implementing {@link CredentialsService}s.
+ * <p>
+ * This base class provides support for receiving <em>Get</em> request messages
+ * via vert.x' event bus and routing them to specific methods accepting the
+ * query parameters contained in the request message.
+ *
+ * @param <T> The type of configuration class this service supports.
+ */
+@Deprecated
```

Comment:
I think adding a `@deprecated` JavaDoc indicating what to use instead would be helpful

Reply:
Honestly I am not sure what to put there. The explanation takes a bit longer, and is summarized in the release notes. Maybe putting a link to the release notes?